src/algorithms/toolbox.jl

```diff
@@ -276,29 +276,25 @@ MPSKit.correlation_length(state, env::CTMRGEnv; num_vals=2, kwargs...) =
     _correlation_length(env; num_vals, kwargs...)
 
 function _correlation_length(env::CTMRGEnv; num_vals=2, kwargs...)
-    T = scalartype(env)
-    ξ_h = Vector{real(T)}(undef, size(env, 2))
-    ξ_v = Vector{real(T)}(undef, size(env, 3))
-    λ_h = Vector{Vector{T}}(undef, size(env, 2))
-    λ_v = Vector{Vector{T}}(undef, size(env, 3))
+    _, n_rows, n_cols = size(env)
 
     # Horizontal
-    λ_h = map(1:size(env, 2)) do r
+    λ_h = map(1:n_rows) do r
         above = InfiniteMPS(env.edges[NORTH, r, :])
-        below = InfiniteMPS(_dag.(env.edges[SOUTH, r, :]))
+        below = InfiniteMPS(_dag.(env.edges[SOUTH, mod1(r + 1, n_rows), :]))
```

Comment:
```suggestion
        below = InfiniteMPS(_dag.(env.edges[SOUTH, _next(r, n_rows), :]))
```
It might look better to use the pre-defined utility function.